### ISSUE TITLE
Support KHR_animation_pointer extension

### ISSUE THREAD

**Issue by issacclee** [vpenades/SharpGLTF#172]
Hi,

I'm looking to implement the support for [KHR_animation_pointer](https://github.com/KhronosGroup/glTF/pull/2147)extension(which is close to getting ratified).

One of the major changes this extension implies is the ability to set animation.channel.target.path as "pointer". I was wondering does that mean we have to add the pointer path definition to the generated gltf.g.cs file and modify the AnimationChannelTarget class to accommodate the new target path type?

**Comment by vpenades** [comment]
Most probably yes.  Another alternative is, given the path seems to be quite "open", I could consider converting it to a string, but then it would break the APIs.

**Comment by vpenades** [comment]
Looking at the extension. The extension itself its tiny, but its implications are huge across all the architecture of glTF

So it's not enough to simply add the extension. After adding it, suddenly, you have to make all the properties that can be targeted by this extension, "animation aware" .... it makes pretty much anything within the scene animatable

**Comment by issacclee** [comment]
> Looking at the extension. The extension itself its tiny, but its implications are huge across all the architecture of glTF
> 
> So it's not enough to simply add the extension. After adding it, suddenly, you have to make all the properties that can be targeted by this extension, "animation aware" .... it makes pretty much anything within the scene animatable

I agree, it will have huge impact on the current implementation. If you take all the necessary validation that needs to be perform into account.

**Comment by vpenades** [comment]
Added support to KHR_Animation_Pointer in multiple commits, most relevant:

https://github.com/vpenades/SharpGLTF/commit/9f71e73320f1b74c66914ea1606f52045e67c88e

https://github.com/vpenades/SharpGLTF/commit/500c20e676602ae2540dff27b74d4faefc642443
